Sources/OktaOAuth2/Authentication/DeviceAuthorizationFlow.swift

```diff
@@ -226,7 +226,37 @@ public final class DeviceAuthorizationFlow: AuthenticationFlow {
     ///   - context: Device authorization context object.
     ///   - completion: Optional completion block for receiving the token, or error result. If `nil`, you may rely upon the ``DeviceAuthorizationFlowDelegate/authentication(flow:received:)`` method instead.
     public func resume(with context: Context, completion: ((Result<Token, APIClientError>) -> Void)? = nil) {
+        self.completion = completion
+        scheduleTimer()
+    }
+    
+    /// Resets the flow for later reuse.
+    public func reset() {
         timer?.cancel()
+        timer = nil
+        context = nil
+        completion = nil
+        isAuthenticating = false
+    }
+
+    // MARK: Private properties / methods
+    static var slowDownInterval: TimeInterval = 5
+    
+    var timer: DispatchSourceTimer?
+    var completion: ((Result<Token, APIClientError>) -> Void)?
+    public let delegateCollection = DelegateCollection<DeviceAuthorizationFlowDelegate>()
```

Comment:
Do you think It might be worth adding a documentation for this since it's a public api?

Reply:
I see your point and I agree with documenting `UsesDelegateCollection`.